newrelic/resource_newrelic_nrql_drop_rule.go

```diff
@@ -0,0 +1,196 @@
+package newrelic
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/newrelic/newrelic-client-go/newrelic"
+	nrErrors "github.com/newrelic/newrelic-client-go/pkg/errors"
+	"github.com/newrelic/newrelic-client-go/pkg/nrqldroprules"
+)
+
+func resourceNewRelicNRQLDropRule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNewRelicNRQLDropRuleCreate,
+		Read:   resourceNewRelicNRQLDropRuleRead,
+		Delete: resourceNewRelicNRQLDropRuleDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"account_id": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Computed:    true,
+				Description: "Account with the NRQL drop rule will be put.",
+			},
+			"action": {
+				Type:         schema.TypeString,
+				ForceNew:     true,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"drop_data", "drop_attributes"}, false),
+				Description:  "The drop rule action (drop_data or drop_attributes).",
+			},
+			"nrql": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+				Description: "Explains which data to apply the drop rule to.",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Optional:    true,
+				Description: "Provides additional information about the rule.",
+			},
+			"rule_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The id, uniquely identifying the rule.",
+			},
+		},
+	}
+}
+
+func resourceNewRelicNRQLDropRuleCreate(d *schema.ResourceData, meta interface{}) error {
+	providerConfig := meta.(*ProviderConfig)
+
+	if !providerConfig.hasNerdGraphCredentials() {
+		return errors.New("err: NerdGraph support not present, but required for Create")
+	}
+
+	client := providerConfig.NewClient
+
+	accountID := d.Get("account_id").(int)
+
+	createInput := []nrqldroprules.NRQLDropRulesCreateDropRuleInput{
+		{
+			Description: d.Get("description").(string),
+			Action:      nrqldroprules.NRQLDropRulesAction(strings.ToUpper(d.Get("action").(string))),
+			NRQL:        d.Get("nrql").(string),
+		},
+	}
+
+	created, err := client.Nrqldroprules.NRQLDropRulesCreate(accountID, createInput)
+	if err != nil {
+		return err
+	}
+
+	rule := created.Successes[0]
+
+	id := fmt.Sprintf("%d:%s", created.Successes[0].AccountID, rule.ID)
```

Comment:
Since you already extract `created.Successes[0]` to `rule`, `created.Successes[0].AccountID` could be `rule.AccountID` which would read easier.